msg/TeamData.msg

```diff
@@ -7,59 +7,42 @@ std_msgs/Header header
 
 # Every value is an array because we can have multiple robots communicating with us.
 # The values match with the robot ids
-uint8[] robot_ids
+uint8 robot_id
 
 uint8 ROLE_IDLING=0
 uint8 ROLE_OTHER=1
 uint8 ROLE_STRIKER=2
 uint8 ROLE_SUPPORTER=3
 uint8 ROLE_DEFENDER=4
 uint8 ROLE_GOALIE=5
-uint8[] role
+uint8 role
 
 uint8 ACTION_UNDEFINED=0
 uint8 ACTION_POSITIONING=1
 uint8 ACTION_GOING_TO_BALL=2
 uint8 ACTION_TRYING_TO_SCORE=3
 uint8 ACTION_WAITING=4
-uint8[] action
+uint8 action
 
 uint8 STATE_INACTIVE=0
 uint8 STATE_ACTIVE=1
 uint8 STATE_PENALIZED=2
-uint8[] state
+uint8 state
 
-# Absolute position values
-geometry_msgs/Pose2D[] robot_positions
+# Absolute position value
+geometry_msgs/Pose robot_position
 
-# Relative ball position, theta of Pose2D is not used
-Position2D[] ball_relative
+# Relative ball position, rotation is not used
+geometry_msgs/PoseWithCovariance ball_relative
 
-# Relative position of the opponent goal, theta of Pose2D is not used
+# Relative position of the opponent goal, rotation is not used
 # This is helpful if the robot has no global position, but sees the goal
-Position2D[] oppgoal_relative
+geometry_msgs/PoseWithCovariance oppgoal_relative
 
-# Positions of opponent robots, if they are recognized
-# The letter of the robot is arbitrary as the sending robot does not know the id of a seen robot
-Position2D[] opponent_robot_a
-Position2D[] opponent_robot_b
-Position2D[] opponent_robot_c
-Position2D[] opponent_robot_d
+ObstacleRelative[] obstacles
```

Comment:
Why don't you use `ObstacleRelativeArray`?

Reply:
Because it is the team data message and i did not care much about it as we need to adapt it later on to the NuBots protocol.